elastic/kibana
# [Fleet] Fix fleet server hosts creation from advanced tab

## Summary

Resolve https://github.com/elastic/kibana/issues/157466

Do not create new fleet server hosts as default when creating them from advanced tab.

That PR is also doing some refacto over that code where a lot of things were dupplicated

## How to test

1. create a default fleet server hosts
2. Add a new host from the advanced tab
<img width="749" alt="Screenshot 2024-01-29 at 9 23 06 AM" src="https://github.com/elastic/kibana/assets/1336873/652b71a2-41cc-4f1c-86f5-5980e19e5043">

3. verify it's not a default host
<img width="1365" alt="Screenshot 2024-01-29 at 9 23 19 AM" src="https://github.com/elastic/kibana/assets/1336873/de9b2550-5ca7-4ba3-9ebd-b93d68ad0d38">
